apps/web/src/views/Swap/V3Swap/index.tsx

```diff
@@ -44,6 +48,22 @@ export function V3SwapForm() {
 
   const insufficientFundCurrency = useCheckInsufficientError(trade)
 
+  const featureEnabled = useExperimentalFeatureEnabled(EXPERIMENTAL_FEATURES.UniversalRouter)
+  const commitButton = useMemo(() => {
+    if (featureEnabled) {
+      return mm?.isMMBetter ? (
+        <MMCommitButton {...mm} />
+      ) : (
+        <SwapCommitButton trade={trade} tradeError={error} tradeLoading={!tradeLoaded} />
+      )
+    }
+    return mm?.isMMBetter ? (
+      <MMCommitButtonV1 {...mm} />
+    ) : (
+      <SwapCommitButtonV1 trade={trade} tradeError={error} tradeLoading={!tradeLoaded} />
+    )
+  }, [error, featureEnabled, mm, trade, tradeLoaded])
```

Comment:
I think it's better to add `React.memo` to component level instead of adding `useMemo` for conditional rendering